File: RegistryPatcher.cs
using Microsoft.Win32;

namespace CrossTalkPatcher;

public static class RegistryPatcher
{
    private const string CrossTalkHost = "ms.msgrsvcs.ctsrv.gay";

    // MSN Messenger < 4.7.2009
    public static void PatchOldMsn()
    {
        using var key = Registry.CurrentUser.CreateSubKey(@"Software\Microsoft\MessengerService");
        key.SetValue("Server", CrossTalkHost, RegistryValueKind.String);
    }

    // MSN/Windows Messenger 4.7.2009 - 3001: only touch the key if it already exists
    public static bool UpdateMsnServerIfPresent()
    {
        using var key = Registry.CurrentUser.OpenSubKey(@"Software\Microsoft\MessengerService", writable: true);
        if (key == null)
            return false;

        key.SetValue("Server", CrossTalkHost, RegistryValueKind.String);
        return true;
    }

    // Yahoo! Messenger 5/6 legacy method (XP SP1 or older)
    public static void PatchLegacyYahoo()
    {
        using var key = Registry.CurrentUser.CreateSubKey(@"Software\Yahoo\Pager");
        key.SetValue("Conn Server", CrossTalkHost, RegistryValueKind.String);
        key.SetValue("socket server", CrossTalkHost, RegistryValueKind.String);
        key.SetValue("Host Name", CrossTalkHost, RegistryValueKind.String);
    }
}

File: registry_patcher.py
# she registry on my windows till i patch
import winreg

CROSSTALK_HOST = "ms.msgrsvcs.ctsrv.gay"

def patch_old_msn():
    """MSN Messenger < 4.7.2009 (registry)"""
    key = winreg.CreateKey(winreg.HKEY_CURRENT_USER, r"Software\Microsoft\MessengerService")
    winreg.SetValueEx(key, "Server", 0, winreg.REG_SZ, CROSSTALK_HOST)
    winreg.CloseKey(key)

def update_msn_server_if_present():
    """MSN/Windows Messenger 4.7.2009 - 3001: only write if the MessengerService key exists"""
    try:
        key = winreg.OpenKey(
            winreg.HKEY_CURRENT_USER,
            r"Software\Microsoft\MessengerService",
            0,
            winreg.KEY_SET_VALUE
        )
        winreg.SetValueEx(key, "Server", 0, winreg.REG_SZ, CROSSTALK_HOST)
        winreg.CloseKey(key)
        return True
    except FileNotFoundError:
        return False

def patch_legacy_yahoo():
    """Yahoo! Messenger 5/6 legacy method (registry)"""
    key = winreg.CreateKey(winreg.HKEY_CURRENT_USER, r"Software\Yahoo\Pager")
    winreg.SetValueEx(key, "Conn Server", 0, winreg.REG_SZ, CROSSTALK_HOST)
    winreg.SetValueEx(key, "socket server", 0, winreg.REG_SZ, CROSSTALK_HOST)
    winreg.SetValueEx(key, "Host Name", 0, winreg.REG_SZ, CROSSTALK_HOST)
    winreg.CloseKey(key)
